amethyst_network/src/utils.rs

```diff
@@ -0,0 +1,72 @@
+extern crate bincode;
+
+use super::{NetConnection, NetConnectionPool, NetEvent, NetSendBuffer, NetSourcedEvent};
+use bincode::{deserialize, serialize, Infinite};
+use bincode::internal::ErrorKind;
+use mio::net::UdpSocket;
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use std::clone::Clone;
+
+/// Sends an event to the target NetConnection using the provided network Socket.
+/// The socket has to be bound.
+pub fn send_event<T>(event: &NetEvent<T>, target: &NetConnection, socket: &UdpSocket)
+where
+    T: Serialize,
+{
+    let ser = serialize(event, Infinite);
+    match ser {
+        Ok(s) => {
+            let slice = s.as_slice();
+            if let Err(e) = socket.send_to(slice, &target.target) {
```

Comment:
Raw socket sending?  No packing of data into the MTU within specified temporal bounds based on how quickly the packet is prioritized?